cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -539,14 +714,26 @@ inline sycl::event fill(sycl::queue& q,
 }
 #endif
 
+/// Multidimensional array
+///
+/// @tparam T           The type of the memory block elements within the multidimensional array.
+///                     :literal:`T` can represent :expr:`float`, :expr:`double` or :expr:`std::int32_t`.
```

Comment:
Is there a restriction on the data type? I can't see anything in this header that suggests the type of data does need to be restricted. Can this comment be removed?